plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -168,6 +173,55 @@ public Version getVersion() throws IOException {
     }
 
     /**
+     * Method returns list of docker containers, include non-running ones.
+     *
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     */
+    public List<ContainerListEntry> listContainers() throws IOException {
+        return listContainers(ListContainersParams.from().withAll(true));
+    }
+
+    /**
+     * Method returns list of docker containers which was filtered by query parameters from {@link ListContainersParams}
+     *
+     * @throws IOException
+     *         in case error parsing response from docker api
+     */
+    public List<ContainerListEntry> listContainers(ListContainersParams params) throws IOException {
+        try (DockerConnection connection = connectionFactory.openConnection(dockerDaemonUri)
+                                                            .method("GET")
+                                                            .path("/containers/json")) {
+            if (params.isAll() != null) {
+                connection.query("all", params.isAll());
+            }
+            if (params.isSize() != null) {
+                connection.query("size", params.isSize());
+            }
+            if (params.getLimit() != null) {
+                connection.query("limit", params.getLimit());
+            }
+            if (params.getSince() != null) {
+                connection.query("since", params.getSince());
+            }
+            if (params.getBefore() != null) {
+                connection.query("before", params.getBefore());
+            }
+            if (params.getFilters() != null) {
+                connection.query("filters", urlPathSegmentEscaper().escape(JsonHelper.toJson(params.getFilters())));
+            }
+            DockerResponse response = connection.request();
+            final int status = response.getStatus();
+            if (OK.getStatusCode() != status) {
+                throw getDockerException(response);
+            }
+            return stream(parseResponseStreamAndClose(response.getInputStream(), ContainerListEntry[].class)).collect(toList());
```

Comment:
It is not efficient to parse json as array and then stream and collect it into list. Use parsing directly into list